### ISSUE TITLE
Feature request - two new options - slide_level, push_headers

### ISSUE THREAD

**Issue by ogobrecht** [neuralyzer/dokuwiki-plugin-revealjs#13]
Hi @neuralyzer,

I implemented the two new options already on my fork of your repo - details below. I am not sure, if you like the naming of the options or the options at all. Please let me know, if I should change something. If you want I can create a pull request.

I created already two pull requests for other things. I am not sure, if you want to have these pull requests to your development or master branch or if you want these requests at all. Please let me know, when I should change anything on these requests.

Best regards
Ottmar
# Changes overview

To make the organization of slide shows more flexible we introduce two new parameters: slide_level and push_headers. For a description see settings.php (next section). An example: I often organize my slideshows with H1 horizontal slides for the main points and then H2 vertical slides for the details. With the existing implementation I am not able to have H2 vertical (nested) slides, because only H3 headers goes vertical. And H3 headers are simply  too small - for the presentation and also for the wiki page - they break the logic, that after a H1 header we normally use a H2 header for the subpoints. By setting the slide level to 1 this is possible. Since all parameters are now overwritable per wiki page (pull request #12), we can adapt this for each presentation. 

If we use slide_level 2 to have H1 and H2 header slides horizontally, then we can push all headers on vertical slides by one with the option push_headers. This can fix our problem with too small slide headers on vertical slides.

At the end we have more flexible ways to organize our slides. In the default settings nothing changes for existing slides - we are backward compatible.
## lang/en/settings.php

``` php
$lang['slide_level'] = "Headers on this level or above starting a horizontal slide. Levels below starting a vertical (nested) slide";
$lang['push_headers'] = "Push headers below slide_level to the next higher one";
```
## conf/metadata.php

``` php
$meta['slide_level'] = array('multichoice','_choices' => array(1, 2));
$meta['push_headers'] = array('onoff');
```
## conf/default.php

``` php
$conf['slide_level'] = 2;
$conf['push_headers'] = 0;
```
## renderer.php
- Function header: implement the two new options slide_level and push_headers
## example_presentation.dokuwiki
- Add the new example options to the start slideshow button
- New testcase for "H3 section following an H1 section"
## README.md
- Add description for the new functionality
- Delete the following warning, because it is in the current implementation not true - An H3 section following an H1 section is currently appended vertically:

```
**Caution**: Only H2 sections open the vertical axis. If an H3 section follows after an H1 section it is appended horizontally.
```

**Comment by neuralyzer** [comment]
Hi there,

The slide_level feature makes immediately sense to me. I'm not sure whether the name slide_level is very descriptive but I also cannot think of a really much better name. I think we could merge this feature as it is.

Your concern regarding H3 headers being too small makes also mush sense. However, I'm not quite sure if pushing the headlines below one level up is the right fix. Pushing headlines up changes the document structure semantically. As I understand, this problem is rather a problem of the presentation of the document. So maybe we should try to fix the CSS instead of changing the headline level? I'm not sure if there is an easy way to do that. If there is an easy way to achieve that, I would prefer changing the CSS, if this turns out to be very cumbersome, the push_headers might be ok.

**Comment by ogobrecht** [comment]
Hi,

you are right - pushing the headers changes the semantics. But the headers
are only pushed on the slides, not in the wiki page, so the semantics on
the wiki page remsins untouched.

With the default slide level 2  I have to use h3 to open the vertical axis.
I have no other possibility. But for me, the normal slide header is a h2
one. By pushing the headers on vertical slides only, I can align all slide
headers to the same optical level. And I can change it for each page.
Again: This is only a optical fix for the slides because we use the headers
as slide indicators - and this leads to some limitations. The slide level
and push headers options give me some possibilities to workaround these
limitations. It can be easily implemented and you do not depend on the css
if something is changed here on the Reveal side in the themes.

Hope you can understand my point.

Best regards
Ottmar

Am 15.09.2016 13:22 schrieb "neuralyzer" <notifications@github.com>:

**Comment by ogobrecht** [comment]
Hi,

I was thinking about it again. At the end you are right - changing the semantics is not the correct way - also on the slides only. I have to think more about this topic and I will only create a pull request for the slide level option. I will do this this evening, because I have currently no access to my computer.

Best regards
Ottmar

**Comment by neuralyzer** [comment]
I'm really not sure either. I see a couple of options/ideas

* use some js to apply h3 styles to h2 styles. But this seems hacky and could possibly break the reveal.js layout. I would prefer not to interact too much with reveal.js
* parse the css files and replace the styles accordingly. This adds complexity to the code.
* do we want to change the font size only or the whole style? We could also overwrite the reveal.js predefined font sizes with inline css in the header. But his does also not seem very nice.
* change the semantics as you suggested. This solution is already there and it works. I'm inclined to actually do it for now as the other options seem more complex.

**Comment by ogobrecht** [comment]
Hi,

after your last comment I decided to leave the feature push_headers in the pull request #14. Feel free to delete the functionality - it is only one line in the configuration files and 4 lines in the header function in the renderer.php and of course the README.md and one parameter in the example presentation button definition. I have currently no better solution for this and as you commented may be all other ideas for this problem are more complex to implement.

Again sorry for the forgotten quotes and the debug call. Four eyes see more than two :-)

Best regards
Ottmar

**Comment by neuralyzer** [comment]
Closed with ogobrecht's commit 0f61937dbf1e9b227bff7156337b31142bda63d1
